src/software/amazon/ion/impl/bin/AbstractIonWriter.java

```diff
@@ -113,8 +113,23 @@ public final void writeValueRecursive(final IonReader reader) throws IOException
                 writeBool(booleanValue);
                 break;
             case INT:
-                final BigInteger bigIntegerValue = reader.bigIntegerValue();
-                writeInt(bigIntegerValue);
+                switch (reader.getIntegerType())
```

Comment:
I can imagine users having to write exactly this structure of code whenever the size matters.  Can we provide a more streamlined approach?  Perhaps a visitor-style callback?